otaclient/app/common.py

```diff
@@ -274,6 +274,26 @@ def re_symlink_atomic(src: Path, target: Union[Path, str]):
             raise
 
 
+def replace_atomic(src: Union[str, Path], dst: Union[str, Path]):
+    """Atomically replace dst file with src file.
+
+    NOTE: atomic is ensured by os.rename/os.replace under the same filesystem.
+    """
+    src, dst = Path(src), Path(dst)
+    if not src.is_file():
+        raise ValueError(f"{src=} is not a regular file or not exist")
+
+    _tmp_file = dst.parent / f".tmp_{os.urandom(6).hex()}"
+    try:
+        # prepare a copy of src file under dst's parent folder
+        shutil.copy(src, _tmp_file, follow_symlinks=True)
```

Comment:
I'd like to add sync operation after shutil.copy.

Reply:
updated in [82f25f0](https://github.com/tier4/ota-client/pull/182/commits/82f25f097ae4ac03129e3201257d0016c680c685).